OData/src/System.Web.OData/OData/Routing/Conventions/ActionRoutingConvention.cs

```diff
@@ -41,7 +41,11 @@ public override string SelectAction(ODataPath odataPath, HttpControllerContext c
                         string actionName = GetAction(odataPath).SelectAction(actionMap, isCollection: false);
                         if (actionName != null)
                         {
-                            controllerContext.AddKeyValueToRouteData(odataPath);
+                            EntitySetPathSegment entitySetPathSegment = (EntitySetPathSegment)odataPath.Segments.First();
+                            IEdmEntityType edmEntityType = entitySetPathSegment.EntitySetBase.EntityType();
+                            KeyValuePathSegment keyValueSegment = (KeyValuePathSegment)odataPath.Segments[1];
+
+                            controllerContext.AddKeyValueToRouteData(keyValueSegment, edmEntityType, ODataRouteConstants.Key);
```

Comment:
Re-lined.

---
In reply to: [63321263](https://github.com/OData/WebApi/pull/727#discussion_r63321263) [](ancestors = 63321263)